docs/standards/smart-contracts/lsp7-digital-asset.md

```diff
@@ -265,6 +265,221 @@ _Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully tra
+
+:::note
+
+#### Requirements:
+
+- `to` cannot be the zero address.
+- `from` cannot be the zero address.
+- `from` must have at least `amount` tokens.
+- If the caller is not `from`, it must be an operator for `from` with access to at least `amount` tokens.
+
+:::
+
+### _burn
+
+```solidity
+function _burn(
+    address from,
+    uint256 amount,
+    bytes memory data
+) internal virtual
+```
+
+Burns `amount` tokens.
+
+_Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully transferred._
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `amount` | uint256  | The amount of token to burn. |
+| `data`   | bytes    | Additional data the caller wants included in the emitted event, and sent in the hook to `from` address. |
+
+:::note
+
+#### Requirements:
+
+- `from` cannot be the zero address.
+- `from` must have at least `amount` tokens.
+- If the caller is not `from`, it must be an operator for `from` with access to at least `amount` tokens.
+
+:::
+
+### _beforeTokenTransfer
+
+```solidity
+function _beforeTokenTransfer(
+    address from,
+    address to,
+    uint256 amount
+) internal virtual
+```
+
+Hook that is called before any token transfer. This includes minting and burning.
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `amount` | uint256  | The amount of token to transfer. |
+
+:::note
+
+#### Requirements:
+
+- When `from` and `to` are both non-zero, ``from``'s `amount` tokens will betransferred to `to`.
+- When `from` is zero, `amount` tokens will be minted for `to`.
+- When `to` is zero, ``from``'s `amount` tokens will be burned.
+
+:::
+
+### _notifyTokenSender
+
+```solidity
+function _notifyTokenSender(
+    address from,
+    address to,
+    uint256 amount,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token sender about the `amount` tokens changing owners using LSP1 interface.
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `amount` | uint256  | The amount of token to transfer. |
+| `data`   | bytes    | Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses. |
+
+### _notifyTokenReceiver
+
+```solidity
+function _notifyTokenReceiver(
+    address from,
+    address to,
+    uint256 amount,
+    bool force,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token receiver about the `amount` tokens changing owners using LSP1 interface. When force is FALSE the token receiver MUST support LSP1.
```

Comment:
An attempt is made to notify the token receiver about the `amount` tokens being received by calling the `universalReceiver(..)` (and link the unviersalReceiver function of the UP) function on the receiver address if it exists.
Sounds better.
Let's also remove this line: `The receiver may revert when the token being sent is not wanted.` because the sender can send and also have a logic where he don't allow any token to be sent.